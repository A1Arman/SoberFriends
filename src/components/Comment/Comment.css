.comment {
    top: 15rem;
}

.comment-modal {
    top: 20rem;
}

.comment-title {
    color: whitesmoke;
}