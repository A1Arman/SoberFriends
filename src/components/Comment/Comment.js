import React, { useState, useEffect } from 'react';
import config from '../../config';
import moment from 'moment';
import TokenService from '../../services/token-service';
import './Comment.css';

const { API_BASE_URL } = config;

function useMergeState(initialState) {
    const [state, setState] = useState(initialState);
    const setMergedState = newState => 
        setState(prevState => Object.assign({}, prevState, newState)
    );
    return [state, setMergedState];
}

function Comment(props) {
    const [commentsRequest, setComments] = useMergeState({
        comments: [],
        isLoading: true
    });

    async function fetchComments() {
        const options = {
            method: 'GET',
            headers: {
                'Content-Type': 'application/json',
                'Authorization': `bearer ${TokenService.getAuthToken()}`
            }
        }
        try {
            const commentResponse = await fetch(`${API_BASE_URL}/comments/${props.postId}`, options);
            const commentsData = await commentResponse.json();
            setComments({
                comments: commentsData,
                isLoading: false
            })
        } catch (err) {
            throw new Error(err)
        }
    }

    useEffect(() => {
        fetchComments();
    }, []);

    const { comments, isLoading } = commentsRequest;


    return (
        <>
        {isLoading ? null : (
            <div className='modal-wrapper comment-modal'
                    style={{
                        transform: props.show ? 'translateY(0vh)' : 'translateY(-100vh)',
                        opacity: props.show ? '1' : '0'
                    }}>
                    <div className='modal-header'>
                        <h3 className='comment-title'>{props.postTitle}</h3>
                    </div>
                    <div className='modal-body'>
                        <header>
                            {comments.map(comment => {
                                return (
                                    <div key={comment.id} className='comment-grid-container'>
                                        <p id='comment-name'>{comment.first_name}</p>
                                        <p id='comment-content'>{comment.comment}</p>
                                        <p id='comment-date'>{moment(comment.date_commented).format('MM.DD.YY')}</p>
                                    </div>
                                )
                            })}
                        </header>
                        <main>
                            <form className='comment-form' id='comment-form' onSubmit={props.handleSubmit}>
                                <div className='grid-container'>
                                    <label htmlFor='comment' id='comment-label'>Comment</label>
                                    <input placeholder='Show your support!' type='text' name='comment' id='comment' required/>
                                </div>
                                <div className="modal-footer">
                                    <button className="btn-cancel" onClick={props.close}><span>CLOSE</span></button>
                                    <button className="btn-continue" type='submit'><span>CONTINUE</span></button>
                                </div>
                            </form>
                        </main>
                    </div>
                </div>
            )}
        </>
    )
}

export default Comment;