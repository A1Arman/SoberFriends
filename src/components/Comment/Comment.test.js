import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Comment from '../Comment/Comment';


describe(`Comment component`, () => {
    it('renders without crashing', () => {
        const wrapper = shallow(<Comment />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});