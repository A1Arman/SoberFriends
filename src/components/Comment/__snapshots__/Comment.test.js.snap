// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment component renders without crashing 1`] = `
<Fragment>
  <Loader />
</Fragment>
`;
